cmd/cloud/dbt.go

```diff
@@ -0,0 +1,150 @@
+
+  $ astro dbt deploy
+`
+
+	dbtDefaultMountPathPrefix = "/usr/local/airflow/dbt/"
+	dbtProjectYmlFilename     = "dbt_project.yml"
+	dbtBundleType             = "dbt"
+)
+
+func newDbtCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "dbt",
+		Short: "Manage your dbt projects deployed to Deployments running on Astronomer",
+	}
+	cmd.AddCommand(
+		newDbtDeployCmd(),
+	)
+	return cmd
+}
+
+func newDbtDeployCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "deploy",
+		Short:   "Deploy your dbt project to a Deployment on Astro",
+		Long:    "Deploy your dbt project to a Deployment on Astro. This command bundles your dbt project files and uploads it to your Deployment.",
+		Args:    cobra.NoArgs,
+		RunE:    deployDbt,
+		Example: dbtDeployExample,
+	}
+
+	cmd.Flags().StringVarP(&mountPath, "mount-path", "m", "", "Path to mount dbt project in Airflow, for reference by DAGs. Default /usr/local/dbt/{dbt project name}")
+	cmd.Flags().StringVarP(&dbtProjectPath, "project-path", "p", "", "Path to the dbt project to deploy. Default current directory")
+	cmd.Flags().StringVar(&workspaceID, "workspace-id", "", "Workspace for your Deployment")
+	cmd.Flags().StringVarP(&deploymentID, "deployment-id", "d", "", "ID of the Deployment to deploy to")
+	cmd.Flags().StringVarP(&deploymentName, "deployment-name", "n", "", "Name of the Deployment to deploy to")
+	cmd.Flags().StringVarP(&deployDescription, "description", "", "", "Description to store on the deploy")
+	cmd.Flags().BoolVarP(&waitForDeploy, "wait", "w", false, "Wait for the Deployment to become healthy before ending the command")
+
+	return cmd
+}
+
+func deployDbt(cmd *cobra.Command, args []string) error {
+	cmd.SilenceUsage = true
+
+	// if the dbt project path is not provided, use the current directory
+	if dbtProjectPath == "" {
+		dbtProjectPath = config.WorkingPath
+	}
+
+	// check that there is a valid dbt project at the dbt project path
+	dbtProjectYamlPath := filepath.Join(dbtProjectPath, dbtProjectYmlFilename)
+	if _, err := os.Stat(dbtProjectYamlPath); os.IsNotExist(err) {
+		return fmt.Errorf("dbt project not found in %s. Please run this command in the root of your dbt project, or use --project-path to specify the dbt project path", dbtProjectYamlPath)
+	}
+	dbtProjectName, err := extractDbtProjectName(dbtProjectYamlPath)
+	if err != nil {
+		return fmt.Errorf("dbt project name not found in %s", dbtProjectYamlPath)
+	}
+
+	// if the workspace ID is not provided, try to find a valid workspace
+	if workspaceID == "" {
+		var err error
+		workspaceID, err = coalesceWorkspace()
+		if err != nil {
+			return fmt.Errorf("failed to find a valid workspace: %w", err)
+		}
+	}
+
+	// if the deployment ID is not provided, get it from the deployment name or prompt the user
+	if deploymentID == "" {
+		selectedDeployment, err := deployment.GetDeployment(workspaceID, "", deploymentName, false, nil, platformCoreClient, astroCoreClient)
```

Comment:
We should do this under cloud/deploy/bundle.go. My understanding of cmd/cloud is for command definitions. Any work specific to the application should be under cloud/ and cmd/cloud should deal with constructing the command based on the flags passed and any pre-checks needed

Reply:
I'd say this part is about the command, because we have multiple ways that the command can specify the deployment:
- The id as an argument
- The name as a flag
- Otherwise from an interactive prompt, which itself can be filtered by another (workspace id) flag

I don't think it is right to pass down a bunch of argument/flag values to the service function and expect deployment resolution there -- that is coupling the application logic of deploying a bundle to the concern that the command that called it has multiple ways of specifying the deployment, and in this case it's even worse than that because it can sometimes involve an interactive prompt.

It should help that line 100 here is passing all the work of talking to the API to an existing service function.

Reply:
You make a good point and i do see an advantage in this, its just my understanding on how i have understood this cobra package. We might need to refactor this in a bunch of different places in the future